### PR TITLE
improve our test Utils a bit and use it at more places

### PR DESCRIPTION
**Pull request by rbri**

### CODE REVIEW

File: tests/src/test/java/org/mozilla/javascript/tests/Utils.java [hunk 1]
```diff
@@ -91,37 +77,44 @@ public static boolean isJavaVersionAtLeast(int desiredVersion) {
...
             final Object expected, final String script) {
+        assertWithAllOptimizationLevels(Context.VERSION_ES6, expected, script);
+    }
+
+    public static void assertWithAllOptimizationLevels(
+            final int languageVersion, final Object expected, final String script) {
         runWithAllOptimizationLevels(
                 cx -> {
-                    cx.setLanguageVersion(Context.VERSION_ES6);
+                    if (languageVersion > -1) {
```
Comment (rPraml) [inline tests/src/test/java/org/mozilla/javascript/tests/Utils.java:97]: I see, `-1` does mean "do not change version" here.

**Review by rPraml — COMMENTED** [review]
looks good to me

File: tests/src/test/java/org/mozilla/javascript/tests/Utils.java (hunk 1 above)
Comment (rbri) [reply tests/src/test/java/org/mozilla/javascript/tests/Utils.java:97]: yes maybe i have to add some comment....

**Comment by gbrail** [comment]
Cool -- thanks!

**Pull request merged**